android-kotlin/app/src/main/java/com/mongodb/app/LoginActivity.kt

```diff
@@ -0,0 +1,108 @@
+        setContentView(R.layout.activity_login)
+        username = findViewById(R.id.input_username)
+        password = findViewById(R.id.input_password)
+
+        loginButton = findViewById(R.id.button_create_login)
+        loginButton.setOnClickListener { onLoginButtonClicked() }
+        loginButton.isEnabled = true
+
+        accountExistsToggleButton = findViewById(R.id.button_existing_account)
+        accountExistsToggleButton.setOnClickListener { toggleCreateUserMode() }
+        accountExistsToggleButton.isEnabled = true
+    }
+
+    /**
+     *  Handle account creation and user authentication
+     */
+    private fun onLoginButtonClicked() {
+        val username = this.username.text.toString()
+        val password = this.password.text.toString()
+        if (isCreateUserMode) {
+            registerUser(username, password)
+        } else {
+           login(username, password)
+        }
+    }
+
+    /**
+     *  Create a new user
+     */
+    private fun registerUser(username: String, password: String) {
+        // while this operation completes, disable the button to login or create a new account
+        loginButton.isEnabled = false
+        // register a user using the Realm App
+        myApp.emailPassword.registerUserAsync(username, password) {
+            // re-enable the button after user registration returns a result
+            loginButton.isEnabled = true
+            if (it.isSuccess) {
+                // when the account has been created successfully, log in to the account
+                Log.i(TAG(), "Successfully registered user.")
+                login(username, password)
+            } else {
+                displayErrorMessage( "Error: ${it.error}")
+            }
+        }
+    }
+
+    /**
+     *  Authenticate a user
+     */
+    private fun login(username: String, password: String) {
+        val creds = Credentials.emailPassword(username, password)
+        myApp.loginAsync(creds) {
+            // re-enable the buttons after user login returns a result
+            loginButton.isEnabled = true
+            if (it.isSuccess) {
+                // on a successful login, open the items screen
+                startActivity(Intent(this, ItemsActivity::class.java))
+            } else {
+                displayErrorMessage( "Error: ${it.error}")
+            }
+        }
+    }
+
+    /**
+     *  Swap between create user mode and log in mode
+     */
+    private fun toggleCreateUserMode() {
+        this.isCreateUserMode = !isCreateUserMode
+        if (isCreateUserMode) {
+            loginButton.setText(getString(R.string.create_account))
+            accountExistsToggleButton.setText(getString(R.string.already_have_account))
+        } else {
+            loginButton.setText(getString(R.string.log_in))
+            accountExistsToggleButton.setText(getString(R.string.does_not_have_account))
+        }
+    }
+
+    /**
+     *  Log an error and propagate it to the user
```

Comment:
Minor, but it's best to prefer simpler language:
```suggestion
     *  Log an error and present it to the user
```

Reply:
fixed